### PR TITLE
Replace `pip-tools` with `poetry`

### PR DESCRIPTION
**Pull request by pamella**
**Description:** Describe in a couple of sentence what this PR adds

- Replace `pip-tools` with `poetry`
  - Remove no longer used files and dependencies
  - Update existing make commands
  - Updating tox config 

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

It could be better to merge this PR once #77 is merged, just in case of code conflicts.

**Merge deadline:** List merge deadline (if any)

N/A

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

- All make commands should be working.
- CI action should be working.
- Tests should be running and passing.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def get_version(*file_paths):
         "Framework ones adding separated serializers for read and write operations"
     ),
     long_description=README + "\n\n" + CHANGELOG,
+    long_description_content_type="text/x-rst",
```
Comment (pamella) [inline setup.py:44]: Drive-by warning fix.

File: .ruff.toml
Comment (pamella) [inline .ruff.toml:1]: Content moved to `pyproject.toml`.

File: .gitignore [hunk 2]
```diff
@@ -63,18 +63,15 @@ docs/modules.rst
...
-# Private requirements
-requirements/private.in
-requirements/private.txt
-
 # tox environment temporary artifacts
 tests/__init__.py
 
 # Development task artifacts
 *.db
+Pipfile
```
Comment (hugobessa) [inline .gitignore]: Is this necessary?

File: tox.ini
Comment (hugobessa) [inline tox.ini:1]: Shouldn't we install only the necessary dependency groups with poetry? This would make the tests run faster and ensure we have only the necessary packages for each type of run.

File: .gitignore (hunk 2 above)
Comment (pamella) [reply .gitignore]: The file is autogenerated during the `missing-migrations` step from `pre-commit`.

Comment (fjsj) [reply .gitignore]: Strange. Maybe pipenv is somehow still installed in your env?

**Review by fjsj — CHANGES_REQUESTED** [review]
Overall looking good, but please check comments

File: tox.ini
Comment (pamella) [reply tox.ini:1]: Good catch. I updated the commands.

File: .gitignore (hunk 2 above)
Comment (pamella) [reply .gitignore]: I will check again.

Comment (pamella) [reply .gitignore]: I pushed a commit with a fix.

**Review by fjsj — APPROVED** [review]

**Review by hugobessa — APPROVED** [review]
